src/packages/react-force-graph-2d/index.d.ts

```diff
@@ -137,16 +141,16 @@ export interface ForceGraphMethods {
   centerAt(x?: number, y?: number, durationMs?: number): ForceGraphKapsuleInstance;
   zoom(): number;
   zoom(scale: number, durationMs?: number): ForceGraphKapsuleInstance;
-  zoomToFit(durationMs?: number, padding?: number, nodeFilter?: (node: NodeObject) => boolean): ForceGraphKapsuleInstance;
+  zoomToFit(durationMs?: number, padding?: number, nodeFilter?: (node: NodeType) => boolean): ForceGraphKapsuleInstance;
 
   // Utility
-  getGraphBbox(nodeFilter?: (node: NodeObject) => boolean): { x: [number, number], y: [number, number] };
+  getGraphBbox(nodeFilter?: (node: NodeType) => boolean): { x: [number, number], y: [number, number] };
   screen2GraphCoords(x: number, y: number): { x: number, y: number };
   graph2ScreenCoords(x: number, y: number): { x: number, y: number };
 }
 
-type FCwithRef<P = {}, R = {}> = React.FunctionComponent<P & { ref?: React.MutableRefObject<R | undefined> }>;
+type FCwithRef = <NodeType extends NodeObject, LinkType extends LinkObject<NodeType>>(props: ForceGraphProps<NodeType, LinkType> & { ref?: React.MutableRefObject<ForceGraphMethods<NodeType, LinkType> | undefined>; }) => React.ReactElement;
```

Comment:
The ideal would be to extend your interface from `NodeObject`

```typescript
interface MyNode extends NodeObject {
  extra: number;
}
```

That way you get `NodeObject`'s types when using `MyNode`.

However, I can see that for people using the package this would be a breaking change.

I have managed to find a way to make the code accept an interface that does not extend from `NodeObject`. This does make the code in the package a bit more difficult to understand but it will allow this release to not be a major release.

Reply:
Fixed in commits https://github.com/vasturiano/react-force-graph/pull/435/commits/f9f40f5f7293876ee646ec7c9b3a570df9d84aae and https://github.com/vasturiano/react-force-graph/pull/435/commits/ec0c2c6211916c440b871b8d8f3fe13b863eef56